packages/now-build-utils/src/fs/run-user-scripts.ts

```diff
@@ -189,6 +189,32 @@ async function scanParentDirs(destPath: string, readPackageJson = false) {
   return { hasPackageLockJson, packageJson };
 }
 
+interface WalkParentDirsProps {
+  start: string;
+  filename: string;
+}
+
+export async function walkParentDirs({
+  start,
+  filename,
+}: WalkParentDirsProps): Promise<string | null> {
+  assert(path.isAbsolute(start));
+  let parent = '';
+
+  for (let current = start; current !== parent; current = parent) {
+    const fullPath = path.join(current, filename);
+
+    // eslint-disable-next-line no-await-in-loop
+    if (await fs.pathExists(fullPath)) {
+      return fullPath;
+    }
+
+    parent = path.dirname(current);
```

Comment:
Note that `dirname()` returns `.` when there is no dirname (i.e. root-level). Not sure if we need to handle that case explicitly here.

Reply:
I don't think that will be possible. I checked and `dirname('/') === '/'`.

I updated the algorithm and added tests. Anything I'm missing?